### PR TITLE
docs: added table of contents and introduction document for Studio tool

### PR DESCRIPTION
**Pull request by Jagrutiti**
**Description**

Added the following document for Studio tool for #552 
- Table of contents
- Introduction

### CODE REVIEW

File: docs/README.md
Comment (Jagrutiti) [inline docs/README.md:1]: Hey @alequetzalli, @derberg , @Florence-Njeri,

I have made this PR and added

**Review by Jagrutiti — COMMENTED** [review]
Hey @alequetzalli, @derberg , @Florence-Njeri,

I have added `Table of Contents` and `Introduction` document in this PR. (I used Generator documentation for guidance)

For `Usage`, I was unable to think of what content to start with. 

Usage is not possible without explaining the Navigation Panel. What if I am not left with unique content for each document.

File: docs/README.md
Comment (alequetzalli) [reply docs/README.md:1]: great start @Jagrutiti! thank you 🙏 

Let's have a quick call to help you with your questions and figure out the next steps. I will DM you now.

**Comment by alequetzalli** [comment]
> For `Usage`, I was unable to think of what content to start with. 
> Usage is not possible without explaining the Navigation Panel. **What if I am not left with unique content for each document.**

lol don't freak out yet @Jagrutiti  ... this to me seems like a possibility for removing `usage` but keeping the other topics in the Table of Contents... thoughts, @magicmatatjahu?

File: docs/introduction.md
Comment (alequetzalli) [inline docs/introduction.md:16]:
```suggestion
To start using Studio, [create an AsyncAPI file](https://www.asyncapi.com/docs/tutorials/streetlights#creating-the-asyncapi-file) for your API.
```

File: docs/introduction.md
Comment (alequetzalli) [inline docs/introduction.md:13]: add bullet point for 'visualizing events flow'

File: docs/introduction.md
Comment (alequetzalli) [inline docs/introduction.md:14]:
```suggestion
# Usage
```

File: docs/introduction.md
Comment (alequetzalli) [inline docs/introduction.md:17]:
```suggestion
<Remember>

If you do not have an `asyncapi.yaml` file ready, then generate one by running: 
`asyncapi new --example=tutorial.yml --no-tty`.

</Remember>
```

File: docs/introduction.md
Comment (alequetzalli) [inline docs/introduction.md:16]:
```suggestion
To start using Studio, [create an AsyncAPI file](https://www.asyncapi.com/docs/tutorials/create-asyncapi-document#create-AsyncAPI-document) for your API. 
```

This link won't work when we merge the new 4 GSoD tutorials to replace the Streetlights tutorial: https://github.com/asyncapi/website/pull/1023/files

File: docs/README.md
Comment (alequetzalli) [inline docs/README.md:4]:
```suggestion
```

File: docs/introduction.md
Comment (alequetzalli) [inline docs/introduction.md:2]: Maybe add more context from studio GH docs?

'One place that allows you to develop an AsyncAPI document, validate it, convert it to the latest version, preview the documentation and visualize the events flow.'

**Comment by magicmatatjahu** [comment]
@alequetzalli @Jagrutiti Sorry for such a delay! Yes, we can now remove `Usage` section and add it in the next PRs. Usage of Studio is hard to describe, because you can use it by CLI (start it), by `studio.asyncapi.com` and by docker - we should think if it's necessary to describe, but we can :)

File: docs/navigation.md
Comment (magicmatatjahu) [inline docs/navigation.md:41]:
```suggestion
The Editor also has resizeable Diagnostics window. It shows a quick count of number of 
- errors (in red),
- warnings (in yellow),
- informations (in blue),
- hints (in green)
by its title. The description of the diagnostics with line number is displayed in a table.
```

File: docs/navigation.md
Comment (magicmatatjahu) [inline docs/navigation.md:49]:
```suggestion
Below the title bar there is a control panel. You can the buttons to:
```

File: docs/navigation.md
Comment (magicmatatjahu) [inline docs/navigation.md:56]:
```suggestion
There is also search bar to search that aids in finding issues by their names. You will see Settings icon to the right. It will take you to the governance of Diagnostic's settings window.
```

File: docs/navigation.md
Comment (magicmatatjahu) [inline docs/navigation.md:86]:
```suggestion
- Simple Example - A basic example of a service that is in charge of processing user signups. Great place to start learning AsyncAPI.
```

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]
I added some suggestions :) Great work @Jagrutiti !

File: docs/navigation.md
Comment (Jagrutiti) [inline docs/navigation.md:26]: Hey @magicmatatjahu 

I have updated the description for `Messages` and `Schemas`.

Do you want me to add the following line to the description as well?

> - we can have it defined but not used by references in the operation fields

**Review by Jagrutiti — COMMENTED** [review]
Hey @magicmatatjahu 

Thank you for reviewing. 

I have added a new document for `settings` and incorporated the changes you suggested.

File: docs/navigation.md
Comment (magicmatatjahu) [reply docs/navigation.md:26]: Yep, you should add this note, that messages and schemas are defined in components but it can (but not must) be used :)

**Comment by Jagrutiti** [comment]
This PR awaits review.

cc: @alequetzalli , @magicmatatjahu

**Comment by alequetzalli** [comment]
Yes, thank you for the reminder @Jagrutiti! 

I am finally able to review this PR, I will finish it by Friday tmw. Thank you for your patience during the past months when it got super busy for me interviewing and onboarding the GSOD writers. 

Excited to continue working with you on this one! ✨😄

**Comment by Jagrutiti** [comment]
Hi @alequetzalli 

I have been selected as a GSoC contributor. I might not be able to work on them for a few more months.

I do not want these PRs to go stale. CC’d you to confirm we are on the same page. Will resume working on them after GSoC.

Take your time to review the PR.

Hope you have a great time with your GSoD contributors.

Looking forward to working with you :)

**Comment by KhudaDad414** [comment]
@Jagrutiti @alequetzalli this PR begs for some attention. 🥺

**Comment by Jagrutiti** [comment]
Review is pending on this PR. I can continue once @alequetzalli  finds time to review.

File: docs/introduction.md
Comment (Amzani) [inline docs/introduction.md:3]: I'll change `develop` here by `create`, as we don't really develop a document.
`convert the document` -> `convert the AsyncAPI document`

File: docs/introduction.md
Comment (Amzani) [inline docs/introduction.md:1]: We could improve that description, It's not just about AsyncAPI documentation.

A `Studio` refer to a development environment / platform, or a collection of tools that enable a certain capability.
IMO I'd define studio as an "AsyncAPI development platform (or environment) that enable users to create, visualise, collaborate on event-driven architectures"

@alequetzalli or @fmvilas might have a better description

File: docs/navigation.md
Comment (Amzani) [inline docs/navigation.md:18]: I suggest to just link or reuse to existing concept definitions as we already have them.

File: docs/navigation.md
Comment (Amzani) [inline docs/navigation.md:36]: `AsyncAPI code` -> `AsyncAPI file`

File: docs/introduction.md
Comment (Amzani) [inline docs/introduction.md:10]: `Realtime` -> `Real-time`

File: docs/navigation.md
Comment (Amzani) [inline docs/navigation.md:73]: Spell out the full term of `USP` Acronym.
e.g `unique selling point (USP)`

We could completely remove this sentence : `Blocks visualizer is the USP of this product` as it's not useful in the documentation context.

File: docs/navigation.md
Comment (Amzani) [inline docs/navigation.md:104]: This sentence can be rephrased or grammatically simplified. typo error : `real-world example templates`

**Review by Amzani — CHANGES_REQUESTED** [review]

File: docs/navigation.md
Comment (Jagrutiti) [reply docs/navigation.md:18]: I could not find the existing definition link.

**Comment by Jagrutiti** [comment]
Hi @Amzani,

Thank you for taking your time to review. 
Except adding the concept definition link (I could not find them) I have made all the changes based on your suggestion.

File: docs/navigation.md
Comment (Amzani) [reply docs/navigation.md:18]: For example server > https://www.asyncapi.com/docs/concepts/server

**Comment by Amzani** [comment]
Thanks @Jagrutiti let's wait for @alequetzalli for final review

File: docs/introduction.md
Comment (alequetzalli) [reply docs/introduction.md:3]: Not sure why Jagruti didn't make this suggested change, but I will do ahead and add it. :)

File: docs/introduction.md
Comment (alequetzalli) [inline docs/introduction.md:3]:
```suggestion
Using Studio you can create an AsyncAPI document, visualize the flow of events, preview, validate, and convert the AsyncAPI document to the latest version in one place.
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:18]:
```suggestion
The Servers section displays the name and information of the [server]( https://www.asyncapi.com/docs/concepts/server) you are using to build your AsyncAPI.
```

Comment (alequetzalli) [reply docs/navigation.md:18]: Adding a link as suggested.

**Review by alequetzalli — COMMENTED** [review]
Almost ready! I made some suggestions based on asks from Samir :)

**Review by alequetzalli — APPROVED** [review]
let's merge thisssss @Amzani 😃

**Comment by Jagrutiti** [comment]
> let's merge thisssss @Amzani 😃

Just went through the comments. Thanks for the approval.

**Review by Amzani — APPROVED** [review]
LGTM

**Review by KhudaDad414 — APPROVED** [review]

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:3]:
```suggestion
This document will walk you through the functions of each button in the navigation panel. 
```

Comment (alequetzalli) [reply docs/navigation.md:3]: I wouldn't call it a guide because it won't be going under the `Guides` content bucket, and would prefer we avoid any confusion 😄

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:9]:
```suggestion
The navigation button is divided into four parts. It provides the following information:
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:26]:
```suggestion
The Messages section displays the messages stored in the `components.messages` section. This does not mean that every message in the component is used.  It can be defined but it may not used by references in the `operation` field.
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:42]:
```suggestion
The Editor also has a resizeable Diagnostics window. It shows a quick count number of... 
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:45]:
```suggestion
- information (in blue),
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:47]:
```suggestion
...by its title. The description of the diagnostics with line number is displayed in a table.
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:55]:
```suggestion
Below the title bar there is a control panel. You can use the buttons to:
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:62]:
```suggestion
There is also a search bar that aids in finding issues by their names. You will see the Settings icon to the right. It will take you to the governance of Diagnostic's settings window.
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:68]:
```suggestion
It starts with displaying the title, version, and description of the API. It renders the formatting too. You can expand each element too via the `key: value` pair of each element, type, payload, and message.
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:75]:
```suggestion
The title card is at the top-left corner of the Block visualizer window. Followed by the Operations card.  You can visualize channels, published messages, payload, type,  patterns, and relationships between the components.
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:81]:
```suggestion
- To the bottom left of the window there are four controls: zoom in, zoom out, full-screen, and toogle interactivity. You zoom the cards in and out or view them in full-screen mode. To prevent the cards from shifting use the toggle interactivity control.
```

File: docs/navigation.md
Comment (alequetzalli) [inline docs/navigation.md:114]:
```suggestion
[You can request to add a Studio template](https://github.com/asyncapi/studio/issues/new?assignees=&labels=enhancement&template=enhancement.md&title=Template%20Request:%20{%20template%20name%20and%20type%20}) to the list if you don't find what you are looking for.
```

File: docs/settings.md
Comment (alequetzalli) [inline docs/settings.md:3]:
```suggestion
You will find the settings icon in the lower-left corner of your screen. It helps you to customize the functioning of the tool to your liking.
```

File: docs/settings.md
Comment (alequetzalli) [inline docs/settings.md:12]:
```suggestion
The editor allows you to:
```

**Review by alequetzalli — COMMENTED** [review]
I wish folks would DM me when they need me for a PR 😥 , when everyone tags me in all PRs, that creates too much noise. Apologies for my delays ❤️ 

This looks about ready to merge, I just submitted the several grammar corrections I noticed needed. But other than the grammar fixes, these docs look amazing and should be merged!

**Comment by alequetzalli** [comment]
Hey there @Jagrutiti ( or @Amzani ? 😸 ), are you able to accept my commit suggestions? I would love to help merge this ❤️

**Comment by alequetzalli** [comment]
@Amzani did you accept my commmit suggestions? It doesn't look like it, thus blocking the merge from happening. :)

**Review by KhudaDad414 — APPROVED** [review]

**Comment by KhudaDad414** [comment]
@magicmatatjahu can't merge this PR without your approval. can you take a look?

**Pull request closed without merging**
